x/posts/client/cli/tx.go

```diff
@@ -67,11 +79,38 @@ func GetCmdCreatePost(cdc *codec.Codec) *cobra.Command {
 				return err
 			}
 
-			msg := types.NewMsgCreatePost(args[1], parentID, allowsComments, args[0], map[string]string{}, from, time.Now().UTC())
+			var msg types.MsgCreatePost
+			msg = types.NewMsgCreateTextPost(
+				args[1],
+				parentID,
+				allowsComments,
+				args[0],
+				map[string]string{},
+				from,
+				time.Now().UTC(),
+			)
+
+			// If there are some medias
+			if len(args) > 3 {
+				var medias types.PostMedias
+				// Read each media and add it to the medias if valid
+				for i := 3; i < len(args); i++ {
+					arg := strings.Split(args[i], ",")
+					if len(arg) == 3 {
+						media := types.NewPostMedia(arg[0], arg[1], arg[2])
+						medias = append(medias, media)
```

Comment:
What would happen if the user specifies two times the same media data? Should we use something like `appendIfMissing` instead of `append` here?

Reply:
I was thinking about that when I was implementing it...it is good to forbid user to post the same media more times? I mean, I can post the same GIF two times,  but that will not be the same for a video or a PDF...so yeah, we should use `appendIfMissing` instead.

Reply:
@bragaz Using `appendIfMissing` here means that we are not allowing to use the same media inside the same post. E.g., the user will not be able to create a post having the same exact GIF, PDF, or file in it. 

This is correct, and does not refer to creating two posts with the same media inside. This means

- Alice creating post P1 with media M1 will be allowed
- Alice creating post P1 with media M1 and M2 will be allowed
- Alice creating post P1 with media M1 and M1 (again) will **not** be allowed
- Alice creating post P1 with media M1 and post P2 with the same media M1 will be allowed